packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -440,8 +538,28 @@ export function makeVirtualReferenceManager(
    * @returns {boolean} true if this possibly creates a GC opportunity
    */
   function ceaseRecognition(vref) {
-    const recognizerSet = vrefRecognizers.get(vref);
     let doMoreGC = false;
+    const { id, facet } = parseVatSlot(vref);
+    if (facet === undefined) {
```

Comment:
`ceaseRecognition()` is only called by `dispatch.retireImports` in liveslots, so it always gets a vref (not a baseref). Moreover, `retireImports` only deals with Presences: there's an assert on `!parseVatSlot(vref).allocatedByVat`. So I don't see how this facet/`facetNames` loop could be correct.

Reply:
@FUDCo explained this to me.. I'd missed a call to `ceaseRecognition()` from `possibleVirtualObjectDeath`, to accomodate the case where an entire cohort is dropped, in which case `ceaseRecognition()` can indeed be called with a baseref instead of a vref. And we confirmed that this clause gets exercised all the time in the new unit tests.

So I'll replace my review with a request to add a comment to `ceaseRecognition()` saying that it could be called with either a vref (in the case of `dispatch.retireImports`) or a baseref (in the case of `possibleVirtualObjectDeath`).

And I'll add a comment (in a different PR) to `dispatch.retireImports` to make it more clear how/when/why it's being called, which should help avoid the sort of confusion we had while discussing this non-problem.